pkg/shares/share_splitting.go

```diff
@@ -126,3 +132,10 @@ func SplitBlobs(cursor int, indexes []uint32, blobs []coretypes.Blob, useShareIn
 	}
 	return writer.Export(), nil
 }
+
+func mergeMaps(mapOne, mapTwo map[coretypes.TxKey]ShareRange) map[coretypes.TxKey]ShareRange {
+	merged := make(map[coretypes.TxKey]ShareRange)
```

Comment:
[non-blocking] we can benefit from increased-performance if we could specify the map capacity at this point i.e., 
```suggestion
	merged := make(map[coretypes.TxKey]ShareRange, len(mapOne)+len(mapTwo))
```